CI tests sometimes raise DuckDuckGoSearchException: 202 Ratelimit

CI tests sometimes raise DuckDuckGoSearchException: 202 Ratelimit

See: https://github.com/huggingface/smolagents/actions/runs/13438187557/job/37545615433?pr=731
> FAILED tests/test_search.py::DuckDuckGoSearchToolTester::test_exact_match_arg - duckduckgo_search.exceptions.DuckDuckGoSearchException: https://lite.duckduckgo.com/lite/ 202 Ratelimit

```
_______________ DuckDuckGoSearchToolTester.test_exact_match_arg ________________

self = <tests.test_search.DuckDuckGoSearchToolTester testMethod=test_exact_match_arg>

    def test_exact_match_arg(self):
>       result = self.tool("Agents")

tests/test_search.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/smolagents/tools.py:190: in __call__
    outputs = self.forward(*args, **kwargs)
src/smolagents/default_tools.py:120: in forward
    results = self.ddgs.text(query, max_results=self.max_results)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <duckduckgo_search.duckduckgo_search.DDGS object at 0x7fa4f14515e0>
keywords = 'Agents', region = 'wt-wt', safesearch = 'moderate', timelimit = None
backend = 'auto', max_results = 10

    def text(
        self,
        keywords: str,
        region: str = "wt-wt",
        safesearch: str = "moderate",
        timelimit: str | None = None,
        backend: str = "auto",
        max_results: int | None = None,
    ) -> list[dict[str, str]]:
        """DuckDuckGo text search. Query params: https://duckduckgo.com/params.
    
        Args:
            keywords: keywords for query.
            region: wt-wt, us-en, uk-en, ru-ru, etc. Defaults to "wt-wt".
            safesearch: on, moderate, off. Defaults to "moderate".
            timelimit: d, w, m, y. Defaults to None.
            backend: auto, html, lite. Defaults to auto.
                auto - try all backends in random order,
                html - collect data from https://html.duckduckgo.com,/
                lite - collect data from https://lite.duckduckgo.com./
            max_results: max number of results. If None, returns results only from the first response. Defaults to None.
    
        Returns:
            List of dictionaries with search results, or None if there was an error.
    
        Raises:
            DuckDuckGoSearchException: Base exception for duckduckgo_search errors.
            RatelimitException: Inherits from DuckDuckGoSearchException, raised for exceeding API request rate limits.
            TimeoutException: Inherits from DuckDuckGoSearchException, raised for API request timeouts.
        """
        if backend in ("api", "ecosia"):
            warnings.warn(f"{backend=} is deprecated, using backend='auto'", stacklevel=2)
            backend = "auto"
        backends = ["html", "lite"] if backend == "auto" else [backend]
        shuffle(backends)
    
        results, err = [], None
        for b in backends:
            try:
                if b == "html":
                    results = self._text_html(keywords, region, timelimit, max_results)
                elif b == "lite":
                    results = self._text_lite(keywords, region, timelimit, max_results)
                return results
            except Exception as ex:
                logger.info(f"Error to search using {b} backend: {ex}")
                err = ex
    
>       raise DuckDuckGoSearchException(err)
E       duckduckgo_search.exceptions.DuckDuckGoSearchException: https://lite.duckduckgo.com/lite/ 202 Ratelimit

.venv/lib/python3.12/site-packages/duckduckgo_search/duckduckgo_search.py:255: DuckDuckGoSearchException
```

--- a/tests/test_search.py
+++ b/tests/test_search.py
@@ -13,7 +13,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import unittest
+
+import pytest
 
 from smolagents import DuckDuckGoSearchTool
 
@@ -25,6 +28,7 @@ class DuckDuckGoSearchToolTester(unittest.TestCase, ToolTesterMixin):
         self.tool = DuckDuckGoSearchTool()
         self.tool.setup()
 
+    @pytest.mark.skipif(not os.getenv("RUN_ALL"), reason="RUN_ALL environment variable not set")
     def test_exact_match_arg(self):
         result = self.tool("Agents")
         assert isinstance(result, str)

